benchmarks/imagenet/resnet50/mocov2.py

```diff
@@ -33,7 +33,11 @@ def __init__(self, batch_size_per_device: int, num_classes: int) -> None:
         self.projection_head = MoCoProjectionHead()
         self.query_backbone = copy.deepcopy(self.backbone)
         self.query_projection_head = MoCoProjectionHead()
-        self.criterion = NTXentLoss(temperature=0.2, memory_bank_size=(65536, 128))
+        self.criterion = NTXentLoss(
+            temperature=0.2,
+            memory_bank_size=(65536, 128),
+            gather_distributed=self.trainer.num_devices > 1,
```

Comment:
This makes sure that the memory bank is identical over all processes (keys are gathered before the memory bank is updated). This follows the original implementation: https://github.com/facebookresearch/moco/blob/1999df86de600027cd2702cbd7cd7b7d2a53559d/moco/builder.py#L68